sql/2020/05_Media/bytes_per_pixel.sql

```diff
@@ -0,0 +1,90 @@
+#standardSQL
+# Bytes per pixel per image format
+CREATE TEMPORARY FUNCTION getImages(payload STRING)
+RETURNS ARRAY<STRUCT<url STRING, naturalWidth INT64, naturalHeight INT64, width INT64, height INT64>> LANGUAGE js AS '''
+try {
+  var $ = JSON.parse(payload);
+  var images = JSON.parse($._Images) || [];
+  return images.map(({url, naturalHeight, naturalWidth, width, height}) => ({url, naturalHeight: Number.parseInt(naturalHeight) || 0, naturalWidth: Number.parseInt(naturalWidth) || 0, width: Number.parseInt(width) || 0, height: Number.parseInt(height) || 0}));
+} catch (e) {}
+return null;
+''';
+
+SELECT
+  a.client,
+  imageType,
+  count(0) as count,
+  APPROX_QUANTILES(if(imageType = 'svg' AND pixels > 0, pixels, naturalPixels), 1000)[OFFSET(100)] AS pixels_p10,
+  APPROX_QUANTILES(if(imageType = 'svg' AND pixels > 0, pixels, naturalPixels), 1000)[OFFSET(250)] AS pixels_p25,
+  APPROX_QUANTILES(if(imageType = 'svg' AND pixels > 0, pixels, naturalPixels), 1000)[OFFSET(500)] AS pixels_p50,
+  APPROX_QUANTILES(if(imageType = 'svg' AND pixels > 0, pixels, naturalPixels), 1000)[OFFSET(750)] AS pixels_p75,
+  APPROX_QUANTILES(if(imageType = 'svg' AND pixels > 0, pixels, naturalPixels), 1000)[OFFSET(900)] AS pixels_p90,
+  APPROX_QUANTILES(bytes, 1000)[OFFSET(100)] AS bytes_p10,
+  APPROX_QUANTILES(bytes, 1000)[OFFSET(250)] AS bytes_p25,
+  APPROX_QUANTILES(bytes, 1000)[OFFSET(500)] AS bytes_p50,
+  APPROX_QUANTILES(bytes, 1000)[OFFSET(750)] AS bytes_p75,
+  APPROX_QUANTILES(bytes, 1000)[OFFSET(900)] AS bytes_p90,
+  APPROX_QUANTILES(round(bytes/if(imageType = 'svg' AND pixels > 0, pixels, naturalPixels), 4), 1000)[OFFSET(100)] AS bpp_p10,
+  APPROX_QUANTILES(round(bytes/if(imageType = 'svg' AND pixels > 0, pixels, naturalPixels), 4), 1000)[OFFSET(250)] AS bpp_p25,
+  APPROX_QUANTILES(round(bytes/if(imageType = 'svg' AND pixels > 0, pixels, naturalPixels), 4), 1000)[OFFSET(500)] AS bpp_p50,
+  APPROX_QUANTILES(round(bytes/if(imageType = 'svg' AND pixels > 0, pixels, naturalPixels), 4), 1000)[OFFSET(750)] AS bpp_p75,
+  APPROX_QUANTILES(round(bytes/if(imageType = 'svg' AND pixels > 0, pixels, naturalPixels), 4), 1000)[OFFSET(900)] AS bpp_p90
+FROM
+(
+  SELECT
+    _TABLE_SUFFIX AS client,
+    p.url AS page,
+    image.url AS url,
+    image.width width,
+    image.height height,
+    image.naturalWidth naturalWidth,
+    image.naturalHeight naturalHeight,
+    ifnull(image.width, 0) * ifnull(image.height, 0) pixels,
+    ifnull(image.naturalWidth, 0) * ifnull(image.naturalHeight, 0) naturalPixels
+  FROM
+    `httparchive.pages.2020_08_01_*` p
+    CROSS JOIN UNNEST(getImages(payload)) AS image
+  WHERE
+    image.naturalHeight > 0
+    AND image.naturalWidth > 0
+--  LIMIT 1000
```

Comment:
Do we need this? If not can we remove?

```suggestion
```

Reply:
@barrypollard I copy pasted the sql from 2019, along with the formatting issues, I will fix it. As for the limit, I guess we can remove this.